File: task02/task02.jl
N = 10^8

a = 3
x = fill(0.1, N)
y = fill(7.1, N)

d = a*x + y

test1 = all(x -> x == 7.4, d)
println("Test 1: $test1")

result = 3*0.1 + 7.1
test2 = all(x -> x == 3*0.1+7.1, d)
println("Test 2: $test2")

File: task02/task02.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    int N = 10;
    double A[N][N], B[N][N], C[N][N];
    bool flag;
    double result;

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            A[i][j] = 3;
            B[i][j] = 7.1;
        }
    }

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            C[i][j] = 0;
            for (int k = 0; k < N; k++) {
                C[i][j] += A[i][k] * B[k][j];
            }
        }
    }

    flag = true;
    result = N * 3 * 7.1;
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (C[i][j] != result) {
                flag = false;
                break;
            }
        }
    }

    if (flag == true) {
        printf("Test: correct\n");
    } else {
        printf("Test: incorrect\n");
    }

}